lib/LTP/utils.pm

```diff
@@ -48,7 +49,18 @@ sub shutdown_ltp {
 }
 
 sub get_ltproot {
-    return get_required_var('TEST_SUITE_NAME') =~ m/[-_]m32$/ ? '/opt/ltp-32' : '/opt/ltp';
+    # TEST_SUITE_NAME is for running 32bit tests (e.g. ltp_syscalls_m32),
+    # checking LTP_PKG is for install_ltp.pm which also uses prepare_ltp_env()
+    my $want_32bit = shift // (get_required_var('TEST_SUITE_NAME') =~ m/[-_]m32$/
+          || get_var('LTP_PKG', '') =~ m/^(ltp|qa_test_ltp)-32bit$/);
+
```

Comment:
Slightly modified `get_ltproot()` here in the second commit to make sure install_ltp does not fail trying to `cd /opt/ltp/...` when installing only 32bit package: http://quasar.suse.cz/tests/6469#step/install_ltp/230 (corner case, normally it shold not happen.